src/msg/async/EventKqueue.cc

```diff
@@ -156,8 +156,13 @@ int KqueueDriver::add_event(int fd, int cur_mask, int add_mask)
     }
   }
   // keep what we set
-  if (fd >= sav_max)
-    resize_events(sav_max+5000);
+  if (fd >= sav_max) {
+    r = resize_events(sav_max+5000);
+    if (r < 0) {
+      lderr(cct) << __func__ << " event count is exceed." << dendl;
+      return r;
```

Comment:
I thought we should report the error to the caller as it is a base component and let the caller do the assert/crash stuff or other error handling. A malloc/realloc failure may not be a good reason to crash. But for sav_events here, maybe we can choose to "crash early".